sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -224,26 +228,27 @@ def check_boston(loss, subsample):
                                         min_samples_split=2,
                                         random_state=1)
 
-        assert_raises(ValueError, clf.predict, boston.data)
-        clf.fit(boston.data, boston.target,
+        assert_raises(ValueError, clf.predict, california.data)
+        clf.fit(california.data, california.target,
                 sample_weight=sample_weight)
-        leaves = clf.apply(boston.data)
-        assert leaves.shape == (506, 100)
+        leaves = clf.apply(california.data)
+        assert leaves.shape == (500, 100)
 
-        y_pred = clf.predict(boston.data)
-        mse = mean_squared_error(boston.target, y_pred)
-        assert mse < 6.0
+        y_pred = clf.predict(california.data)
+        mse = mean_squared_error(california.target, y_pred)
+        assert mse < 0.1
 
         if last_y_pred is not None:
-            assert_array_almost_equal(last_y_pred, y_pred)
+            assert_array_almost_equal(last_y_pred, y_pred, decimal=0)
```

Comment:
Before to merge we need to add a FIXME to explain the issue and link to the relevant PR/issue just to not lose track

```suggestion
            # FIXME: `decimal=0` is very permissive. This is due to the fact that
            # GBRT with and without `sample_weight` do not use the same implementation
            # of the median during the initialization with the `DummyRegressor`.
            # In the future, we should make sure that both implementations should be the same.
            # Refer to https://github.com/scikit-learn/scikit-learn/pull/17377 for some detailed
            # explanations.
            assert_array_almost_equal(last_y_pred, y_pred, decimal=0)
```